### ISSUE TITLE
macOS 10.15 Catalina

### ISSUE THREAD

**Issue by big610** [FreeTubeApp/FreeTube#409]
Using macOS 10.15 Catalina FreeTube 0.71 would not open. Put back FreeTube 0.70 which had run fine on Catalina, now also would not open. Restored just FreeTube 0.70 from Time Machine Back Up from Mojave, able to run FreeTube 0.70 again. Best of luck with Catalina.

**Comment by wjyrf8aqz** [comment]
Same issue with Catalina, thanks for sharing the Time Machine idea. I didn't even think about that, but it got FreeTube working again!

**Comment by big610** [comment]
Preston when you make time, please explain to us in full detail, what happened, so we can learn.

**Comment by PrestonN** [comment]
I'm going to copy and paste what I sent to someone through an email a while ago.

> It's something I need to look more into, but I believe it's related to the below articles.


https://www.macrumors.com/guide/32-bit-mac-apps/

https://www.macrumors.com/2019/10/03/apple-macos-catalina-mac-app-store/


Right now FreeTube is built off of my Ubuntu machine for all versions, including Mac.  It seems that my only option for fixing this is to start doing Mac builds on an actual Mac, which I do not own.  I will have to see what my options are for obtaining a Mac or if there is a cheaper way for me to make proper builds again.  I will likely have a blog posted when this is ready.


If you know what you're doing and comfortable with the command line, you can try building from source using the instructions below, which may fix your problem.


https://github.com/FreeTubeApp/FreeTube/wiki/Environment-Setup-and-Packaging


To add on to what I said, it's possible some CI build tools could fix it as well, but I need to look into it.

**Comment by big610** [comment]
I looked at your all your links. Best guess is something done on Ubuntu, then CI.
Perhaps you can overcome it.

I CodeSigned, did not fix, other can try.
The app will say is damaged, yours did not, I wish I wrote down what happened.
If somebody else can write down in complete detail word for word, what happens opening FreeTube on Catalina.

CodeSign

In Terminal 
Type:sudo codesign -f -s -
Press: space key
Drag & Drop FreeTube app from Applications into Terminal
Press: return key
Type: your password
Press: return key

Deep Method:
codesign --sign - --force --deep /Applications/FreeTube.app

Your next Beta Version needs to be CodeSigned, that can be a start.
Let me know I'll test it for you, since I have a Time Machine Backup from Mojave.

Hackintosh is another option, Preston to research.
If and when you start over backing up, buy an enclosure to clone what you have on Ubuntu.
I can help you more in detail when ready, what the best Enclosure & Cable are.
I use three methods Time Machine to External, Enclosure & SSD with SuperDuper!, Copy & Paste to External.

**Comment by big610** [comment]
If the Catalina issue was Notarization, I would have had to notarize FreeTube, did not need it.

Catalina on Ubuntu, needs research.

**Comment by PrestonN** [comment]
I have borrowed a Mac with Catalina from someone to take a look into this problem.  It seems to throw an error message when you first try to open it but if you right click to open it then FreeTube should open up on your machine.  This article from Apple explains more on how to open an unverified app.

https://support.apple.com/en-ph/HT202491

I've mentioned this before, but I'd like to avoid code signing for now if I can. The process is very expensive and I'd have to make a similar cost for Windows apps.  The work around seems simple enough to not require any effort from me so I don't see a reason to push out a fix for now.  If the problem gets worse I will consider code signing but for now it seems to be fine without it.

**Comment by big610** [comment]
Both 0.7.0 & 0.71, I get the mesage: The application “FreeTube” can’t be opened. 
From Applications, Right Click Open, same message.
Different message that your support link.

If it needed code signing, the deep sumo commands would fix it to open. Those commands do nothing, still remains: The application “FreeTube” can’t be opened.
Confident it is not a Code Sign issue, or I could have fixed it in Terminal with:
codesign --sign - --force --deep /Applications/FreeTube.app

**Comment by PrestonN** [comment]
I've added a `.dmg` build over on the [releases](https://github.com/FreeTubeApp/FreeTube/releases/tag/v0.7.1-beta) section.  Please download this and give it a try on your system.  I was able to test that this indeed works on a Catalina based VM where the current release gives the error that has been mentioned.  If this works on your systems, then I will update the website and use `.dmg` for all future builds instead of the `.zip` format.

**Comment by big610** [comment]
FreeTube-0.7.1_mac.dmg
November 9 2019

Opens & runs, macOS Catalina 10.15.1.
Thank you. So we can all learn what was where the issues that caused the problem?
.dmg should always be the only upload for mac.

**Comment by PrestonN** [comment]
To be truthful, I had a difficult time figuring out why the .zip release was causing issues.  I decided to build the project to a .dmg format to see if that would fix the problem and that seemed to do it.  I didn't do anything else to make it work.  I never had access to a Mac which is why I never provided a .dmg release to begin with.  Since I was able to get a VM setup properly and the build works across all versions, I will simply use those builds for future releases instead of the .zip format.  I believe most Mac users prefer .dmg anyways so I don't think anyone will be too upset with this change.

Thank you for the recommendation to zip up the .exe and .dmg files.  I will likely do that in the next release.

Since this seems to be finally fixed up, I will close this issue.  Expect a new blog post with an announcement to the new build along with the website updating to reflect the changes as well.
